main.go

```diff
@@ -0,0 +1,235 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type StatusChecker interface {
+	Check(ctx context.Context, name string) (status bool, err error)
+}
+
+type httpChecker struct {
+}
+
+/*
+Function having context working as client to fetch status of websit
+*/
+
+func (h httpChecker) Check(ctx context.Context, sitename string) (status bool,
+	err error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	req, err := http.NewRequest(http.MethodGet, sitename, nil)
+	if err != nil {
+		log.Println(err)
+	}
+	req = req.WithContext(ctx)
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false, err
+	}
+	if res.StatusCode != http.StatusOK {
+		return false, err
+	}
+	return true, nil
+}
+
+/*
+go routines working as a server context with timeout of 1 minute
+*/
+func checkSites() {
+	log.Println("STATUS CHECK STARTED")
+	ctxAsClient := context.Background()
+	httpChecker := httpChecker{}
+
+	for {
+		select {
+		case <-time.After(1 * time.Minute):
+			log.Println("STATUC CHECK :", websitesMap)
+
+			for site := range websitesMap {
+				status, err := httpChecker.Check(ctxAsClient, site)
+				if err != nil || !status {
+					websitesMap[site] = "DOWN"
+				} else {
+					websitesMap[site] = "UP"
+				}
+			}
+		case <-ctxAsClient.Done():
+			err := ctxAsClient.Err()
```

Comment:
What happens if the request takes more than 1 min ?

Reply:
If it takes more than a minute then we are marking it as done.
It's not in a requirements, but I have added it to clear understanding on context.